src/System.Private.ServiceModel/src/System/ServiceModel/Channels/SslStreamSecurityUpgradeProvider.cs

```diff
@@ -226,6 +226,20 @@ protected override void OnClose(TimeSpan timeout)
             CleanupServerCertificate();
         }
 
+        protected internal override Task OnCloseAsync(TimeSpan timeout)
+        {
+            return OnCloseAsyncInternal(timeout);
+        }
+
+        private async Task OnCloseAsyncInternal(TimeSpan timeout)
+        {
+            if (_clientCertificateAuthenticator != null)
+            {
+                await Task.Run(() => SecurityUtils.CloseTokenAuthenticatorIfRequired(_clientCertificateAuthenticator, timeout));
+            }
+            CleanupServerCertificate();
+        }
```

Comment:
There's no need to do anything async artificially. Just do everything synchronously and return Task.CompletedTask